webapp/channels/src/components/post/index.tsx

```diff
@@ -147,7 +150,9 @@ function makeMapStateToProps() {
             teamName = team?.name || currentTeam?.name;
         }
 
-        const canReply = isDMorGM || (channel.team_id === currentTeam?.id);
+        const burnOnReadFeatureEnabled = isBurnOnReadEnabled(state);
+        const isBurnOnReadPost = burnOnReadFeatureEnabled && post.type === PostTypes.BURN_ON_READ;
```

Comment:
Similar to a previous comment, we should either use the isBurnOnReadPost selector, or remove that selector and use direct type checks like here.